cmd/swapd/contract.go

```diff
@@ -74,53 +70,25 @@ func deploySwapFactory(
 	forwarderAddress ethcommon.Address,
 	dataDir string,
 ) (ethcommon.Address, *contracts.SwapFactory, error) {
+
 	if privkey == nil {
 		return ethcommon.Address{}, nil, errNoEthereumPrivateKey
 	}
 
-	chainID, err := ec.ChainID(ctx)
-	if err != nil {
-		return ethcommon.Address{}, nil, err
-	}
-	txOpts, err := bind.NewKeyedTransactorWithChainID(privkey, chainID)
-	if err != nil {
-		return ethcommon.Address{}, nil, fmt.Errorf("failed to make transactor: %w", err)
-	}
-
 	if (forwarderAddress == ethcommon.Address{}) {
 		// deploy forwarder contract as well
-		address, err := deployForwarder(ctx, ec, txOpts) //nolint:govet
```

Comment:
The removed code from this file was just moved to the `ethereum` folder (`contracts` package). I needed it for writing tests in that package. The parts that seemed like application specific logic were left here:
* Returning an error if the eth key was not passed (the version in the contracts package assumes nil was not passed)
* Deciding whether to deploy  a forwarder if none was set (`DeploySwapFactory` in the `ethereum` folder will deploy the contract with an all zero forwarder if that's what you ask it to do)
* Writing out the contract address to the file system